src/views.jl

```diff
@@ -0,0 +1,41 @@
+# Permit Base.view for Dataset and Attribute to normalize Base.copyto! syntax
+
+# Long term consideration:
+# If Dataset and Attribute become an AbstractArray, perhaps these should just be SubArrays ?
+"""
+    DatasetView
+
+This is an experimental data type and may change. Use `view` rather than `DatasetView` directly.
+"""
+struct DatasetView
+    parent::Dataset
+    indices
```

Comment:
Would it make sense to store the dataspace?

Reply:
I think this is worthwhile eventually, but I think we should defer this to a larger overhaul of `Dataset`.

In particular, we should consider parameterizing `Dataset` and making it an `AbstractArray` or `AbstractDiskArray` for version 0.17.

Reply:
See #930

Reply:
I'm leaning towards removing the view feature from this pull request.

Reply:
I think it's a good idea, but I would suggest making it actually a functional part: i.e. calling `read!(dataset, array, I...)` would internally call `read!(view(dataset, I...), array)`

Reply:
I agree. I just think the `view` component is much more complicated than the other parts.